lib/tasks/serve.js

```diff
@@ -57,9 +81,20 @@ class ServeTask extends Task {
     this._runDeferred = RSVP.defer();
 
     return expressServer.start(options)
-      .then(() => this._runDeferred.promise);
+      .then(() => {
+        if (hasBuild) {
+          this.serveUrl(options);
```

Comment:
This surprises me, don't we still want to share the serve url? The user will still visit this URL, but we merely wont do rebuilds for them automatically.